util/type_traits.h

```diff
@@ -53,9 +53,8 @@ template <typename T>
 using is_cl_float_type =
     std::bool_constant<std::is_floating_point<T>::value ||
                        std::is_same<sycl::half, T>::value ||
-                       std::is_same<sycl::cl_float, T>::value ||
-                       std::is_same<sycl::cl_double, T>::value ||
-                       std::is_same<sycl::cl_half, T>::value>;
+                       std::is_same<float, T>::value ||
+                       std::is_same<double, T>::value>::value>;
```

Comment:
Aren't these types already covered by `std::is_floating_point<T>::value` check?

Reply:
They are. Removed double and float check.